src/main/java/com/vi/migrationtool/tenantservice/DiocesesToTenantsMigrator.java

```diff
@@ -0,0 +1,95 @@
+package com.vi.migrationtool.tenantservice;
+
+import com.vi.migrationtool.common.MigrationTasks;
+import com.vi.migrationtool.config.BeanAwareSpringLiquibase;
+import java.util.List;
+import java.util.stream.Collectors;
+import liquibase.database.Database;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@Slf4j
+public class DiocesesToTenantsMigrator extends MigrationTasks {
+
+  private static final String dioceseTableName = "diocese";
+
+  @Builder
+  static class Diocese {
+    private String name;
+  }
+
+  @Builder
+  static class Tenant {
+    private String name;
+  }
+
+  @Override
+  public void execute(Database database) {
+    log.info("Starting import of dioceses from agency to tenant");
+    val agencyJdbcTemplate =
+        BeanAwareSpringLiquibase.getNamedBean("agencyServiceJdbcTemplate", JdbcTemplate.class);
+
+    var dioceseTableExist = doesDioceseTableExist(agencyJdbcTemplate);
+    if (!dioceseTableExist) {
+      log.info("diocese table does not exist, skipping import");
+      return;
+    }
+    val dioceses = getAllDiocesesFromAgencyDB(agencyJdbcTemplate);
+
+    log.info("All dioceses: {}", dioceses);
+
+    var tenantServiceJdbcTemplate =
+        BeanAwareSpringLiquibase.getNamedBean("tenantServiceJdbcTemplate", JdbcTemplate.class);
+
+    val existingTenants = getAllTenantsFromTenantsDB(tenantServiceJdbcTemplate);
+
+    val newTenants = getNewTenants(dioceses, existingTenants);
+
+    log.info("New tenants: {}", newTenants);
+
+    val result = insertNewTenants(newTenants, tenantServiceJdbcTemplate);
+
+    log.info("Inserted {} new tenants", result.length);
+    log.info("Finished migration of dioceses to tenants");
+  }
+
+  private static int[] insertNewTenants(
```

Comment:
hm, now I see another issue with this approach. 
if you create such tenant with only two attributes subdomain and name it's not gonna work. I mean, maybe you will create a tenant in DB, but this not a complete tenant object. So if you try to actually access it in the app, it will crash.
I think in this case the easiest way is to call tenantService admin API to create new tenant. Because it will create this tenant and all default tenant settings (including the part that still resides in consulting types).